Sources/SwiftProtobuf/AnyMessageStorage.swift

```diff
@@ -301,163 +323,173 @@ extension AnyMessageStorage {
+            return false
+      }
 }
 
 // _CustomJSONCodable support for Google_Protobuf_Any
 extension AnyMessageStorage {
-  // Override the traversal-based JSON encoding
-  // This builds an Any JSON representation from one of:
-  //  * The message we were initialized with,
-  //  * The JSON fields we last deserialized, or
-  //  * The protobuf field we were deserialized from.
-  // The last case requires locating the type, deserializing
-  // into an object, then reserializing back to JSON.
-  func encodedJSONString(options: JSONEncodingOptions) throws -> String {
-    switch state {
-    case .binary(let valueData):
-      // Transcode by decoding the binary data to a message object
-      // and then recode back into JSON.
-      guard let messageType = Google_Protobuf_Any.messageType(forTypeURL: _typeURL) else {
-        // If we don't have the type available, we can't decode the
-        // binary value, so we're stuck.  (The Google spec does not
-        // provide a way to just package the binary value for someone
-        // else to decode later.)
-        throw JSONEncodingError.anyTranscodeFailure
+      // Override the traversal-based JSON encoding
+      // This builds an Any JSON representation from one of:
+      //  * The message we were initialized with,
+      //  * The JSON fields we last deserialized, or
+      //  * The protobuf field we were deserialized from.
+      // The last case requires locating the type, deserializing
+      // into an object, then reserializing back to JSON.
+      func encodedJSONString(options: JSONEncodingOptions) throws -> String {
+            switch state {
+            case .binary(let valueData):
+                  // Transcode by decoding the binary data to a message object
+                  // and then recode back into JSON.
+                  guard
+                        let messageType = Google_Protobuf_Any.messageType(
+                        forTypeURL: _typeURL)
+                  else {
+                        // If we don't have the type available, we can't decode the
+                        // binary value, so we're stuck.  (The Google spec does not
+                        // provide a way to just package the binary value for someone
+                        // else to decode later.)
+                        throw JSONEncodingError.anyTranscodeFailure
+                  }
+                  let m = try messageType.init(
+                        serializedData: valueData, partial: true)
+                  return try serializeAnyJSON(
+                        for: m, typeURL: _typeURL, options: options)
+
+            case .message(let msg):
+                  // We should have been initialized with a typeURL, but
+                  // ensure it wasn't cleared.
+                  let url = !_typeURL.isEmpty
+                        ? _typeURL : buildTypeURL(
+                              forMessage: msg,
```

Comment:
I'm wondering what would happen here if the expression after `:` was a long string of additions instead of a function call (aside from warnings about typechecker performance, I mean ;-) )